volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -43,104 +69,222 @@ def _generator(self):
-                            format_hints.Hex(attr_data.vol.offset),
-                            mft_record.get_signature(),
-                            mft_record.RecordNumber,
-                            mft_record.LinkCount,
-                            mft_flag,
-                            renderers.NotApplicableValue(),
-                            attr.Attr_Header.AttrType.lookup(),
-                            conversion.wintime_to_datetime(attr_data.CreationTime),
-                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
-                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
-                            conversion.wintime_to_datetime(attr_data.AccessedTime),
-                            renderers.NotApplicableValue(),
-                        )
-
-                    # File Name Attribute
-                    if attr.Attr_Header.AttrType.lookup() == "FILE_NAME":
-                        attr_data = attr.Attr_Data.cast(fn_object)
-                        file_name = attr_data.get_full_name()
-
-                        # If we don't have a valid enum, coerce to hex so we can keep the record
-                        try:
-                            permissions = attr_data.Flags.lookup()
-                        except ValueError:
-                            permissions = hex(attr_data.Flags)
-
-                        yield 1, (
-                            format_hints.Hex(attr_data.vol.offset),
-                            mft_record.get_signature(),
-                            mft_record.RecordNumber,
-                            mft_record.LinkCount,
-                            mft_flag,
-                            permissions,
-                            attr.Attr_Header.AttrType.lookup(),
-                            conversion.wintime_to_datetime(attr_data.CreationTime),
-                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
-                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
-                            conversion.wintime_to_datetime(attr_data.AccessedTime),
-                            file_name,
-                        )
+                    for record in attr_callback(mft_record, attr, symbol_table):
+                        yield record
 
                     # If there's no advancement the loop will never end, so break it now
                     if attr.Attr_Header.Length == 0:
                         break
 
                     # Update the base offset to point to the next attribute
                     attr_base_offset += attr.Attr_Header.Length
-                    attr = self.context.object(
+                    # Get the next attribute
+                    attr = context.object(
                         attribute_object,
                         offset=offset + attr_base_offset,
                         layer_name=layer.name,
                     )
 
+    @staticmethod
```

Comment:
Happy with these as static methods, we just need to remember they count as part of the external API if they ever change in the future, they should cause the version number to bump too.